### PR TITLE
Fixes #88

### LINKED ISSUE #88: 2x your capapility drops!
Currently, autofix does not detect that caps have already been dropeed, so it drops them again.
I haven't had a look at why, but this is the result:

```yaml
          capabilities:
            drop:
            - AUDIT_WRITE
            - CHOWN
            - DAC_OVERRIDE
            - FOWNER
            - FSETID
            - KILL
            - MKNOD
            - NET_BIND_SERVICE
            - SETGID
            - SETFCAP
            - SETPCAP
            - SETUID
            - SYS_CHROOT
            - AUDIT_WRITE
            - CHOWN
            - DAC_OVERRIDE
            - FOWNER
            - FSETID
            - KILL
            - MKNOD
            - NET_BIND_SERVICE
            - NET_RAW
            - SETFCAP
            - SETGID
            - SETPCAP
            - SETUID
            - SYS_CHROOT
```

### PR DESCRIPTION
**Pull request by klautcomputing**
Kubeaudit now correctly audits all containers.

closes #88

### CODE REVIEW

File: cmd/capabilities_fixes.go [hunk 1]
```diff
@@ -14,31 +14,32 @@ func fixCapabilitiesNIL(resource k8sRuntime.Object) k8sRuntime.Object {
...
 }
 
 func fixCapabilityNotDropped(resource k8sRuntime.Object, occurrence Occurrence) k8sRuntime.Object {
-	var containers []Container
-	for _, container := range getContainers(resource) {
-		container.SecurityContext.Capabilities.Drop = append(container.SecurityContext.Capabilities.Drop, Capability(occurrence.metadata["CapName"]))
-		containers = append(containers, container)
+	containers := getContainers(resource)
+	for _, container := range containers {
+		if occurrence.container == container.Name {
```
Comment (jinankjain) [inline cmd/capabilities_fixes.go]: Is container.name unique for every container inside a k8s resource?

Comment (fedorlis) [reply cmd/capabilities_fixes.go]: According to https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/, names are unique within namespaces but not necessarily across them.

Comment (klautcomputing) [reply cmd/capabilities_fixes.go]: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2089-L2093 So from what I take from the comment they should be unique.

**Comment by klautcomputing** [comment]
As @jinankjain pointed out the code currently parses invalid `yaml` files incorrectly instead of erroring out.
I don't think that's really in our realm to address general validation of templates. I really like the Unix philosophy of: *Do one thing do it well*. In our case that would be address security issues. If the config is not correct that should be caught by a validator.
But I will patch a quick check in there that names need to be unique :)

File: cmd/util_test.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDoubleName(t *testing.T) {
```
Comment (fedorlis) [inline cmd/util_test.go]: ❤️

**Comment by klautcomputing** [comment]
#84

**Pull request merged**